<a href="https://colab.research.google.com/github/mennareda123/flyrank-ml-journey/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I chose the Content Refresh Prioritization lane because websites can contain thousands of pages, making it difficult for SEO teams to review every page manually. By identifying the pages that should be refreshed first, the team can focus its limited time and effort where it is most valuable. This project supports better decision-making rather than simply building a machine learning model.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Decision
Help the SEO team decide which web pages should be refreshed first.

### Action
The SEO team reviews and updates the highest-priority pages before spending time on lower-priority pages.

### Unit of Analysis:
A single web page.


### Cost of a wrong recommendation
A wrong recommendation can waste time and resources on pages that do not need updates while delaying improvements to pages that actually need attention. This may lead to continued traffic decline.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")
df.head()

,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [ ]:
df.shape

(30000, 44)

In [ ]:
df["trend_direction"].value_counts()

,count
trend_direction,
down,16262
stable,5962
up,4388
new,2236
flat,1152


In [ ]:
df["client_id"].nunique()

32

The starter dataset contains **30,000 web pages** described by **44 features**.

The data comes from **32 different clients**, making it more diverse.

More than half of the pages (**16,262 out of 30,000**) are labeled as **down**, suggesting that prioritization could help SEO teams focus their effort where it is most needed.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project provides decision support by identifying pages that may need attention based on observed patterns in the data. The results should help the SEO team prioritize their work, not replace human judgment.

This project does not prove why a page's traffic changes, predict Google's ranking algorithm, or guarantee that refreshing a page will improve its performance.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.

In [ ]:
!git clone https://github.com/mennareda123/flyrank-ml-journey.git

Cloning into 'flyrank-ml-journey'...
remote: Enumerating objects: 98, done.
remote: Counting objects: 100% (98/98), done.
remote: Compressing objects: 100% (69/69), done.
remote: Total 98 (delta 20), reused 80 (delta 13), pack-reused 0 (from 0)
Receiving objects: 100% (98/98), 1.83 MiB | 11.82 MiB/s, done.
Resolving deltas: 100% (20/20), done.


In [ ]:
%cd flyrank-ml-journey

/content/flyrank-ml-journey/flyrank-ml-journey


In [ ]:
!ls

AGENTS.md  DATA_USE.md	LICENSE    README.md	     SETUP.md	 work
CLAUDE.md  docs		notebooks  requirements.txt  skills
data	   GUIDE.md	outputs    scripts	     submission
